src/agent/rustjail/src/capabilities.rs

```diff
@@ -126,13 +126,13 @@ pub fn drop_privileges(cfd_log: RawFd, caps: &LinuxCapabilities) -> Result<()> {
     )
     .map_err(|e| anyhow!(e.to_string()))?;
 
-    if let Err(_) = caps::set(
+    caps::set(
         None,
         CapSet::Ambient,
         to_capshashset(cfd_log, caps.ambient.as_ref()),
-    ) {
-        log_child!(cfd_log, "failed to set ambient capability");
-    }
+    )
+    .map_err(|_| log_child!(cfd_log, "failed to set ambient capability"))
```

Comment:
I appreciate this is "in keeping" with the original code, but I'd prefer we also log the error as it might help us understand what went wrong :)

Reply:
I want to keep the logic same as before.

Reply:
I think the `ok()` is something "trick", on the other hand `if let Err()` is easy to understand.

Reply:
I have replace the `ok()` with `let _ = `